### PR TITLE
feat(cb2-8218): list of deleted reference data

### PR DESCRIPTION
**Pull request by emilyrkelly21**
## List of deleted Reference Data

Ability to view all Reference Data that has been deleted
[CB2-8218](https://dvsa.atlassian.net/browse/CB2-8218)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

**Review by owen-corrigan-bjss — DISMISSED** [review]

File: src/app/features/reference-data/reference-data-deleted-list/reference-data-deleted-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  get refDataAdminType$(): Observable<any | undefined> {
+    return this.store.pipe(select(selectReferenceDataByResourceKey(ReferenceDataResourceType.ReferenceDataAdminType, this.type)));
+  }
+
+  get data$(): Observable<ReferenceDataModelBase[] | undefined> {
+    return this.store.pipe(select(selectSearchReturn((this.type + '#AUDIT') as ReferenceDataResourceType)));
+  }
+
+  public get roles(): typeof Roles {
```
Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-deleted-list/reference-data-deleted-list.component.ts:43]:
```suggestion
  get roles(): typeof Roles {
```

File: src/app/features/reference-data/reference-data-list/reference-data-list.component.ts [hunk 2]
```diff
@@ -53,6 +93,12 @@ export class ReferenceDataListComponent implements OnInit {
     });
   }
 
+  deletedItems(): void {
```
Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-list/reference-data-list.component.ts:96]:
```suggestion
  navigateToDeletedItems(): void {
```

File: src/app/features/reference-data/reference-data-list/reference-data-list.component.ts [hunk 3]
```diff
@@ -53,6 +93,12 @@ export class ReferenceDataListComponent implements OnInit {
...
+    this.router.navigate(['deleted-items'], { relativeTo: this.route }).then(() => {
+      window.location.reload();
```
Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-list/reference-data-list.component.ts:98]: why are we doing a refreshing the page?

**Review by owen-corrigan-bjss — APPROVED** [review]

File: src/app/features/reference-data/reference-data-list/reference-data-list.component.ts (hunk 3 above)
Comment (emilyrkelly21) [reply src/app/features/reference-data/reference-data-list/reference-data-list.component.ts:98]: I've taken this out now, I had just did it the same as the amend and add, but I realise we don't actually need it for the deleted page

File: src/app/features/reference-data/reference-data-list/reference-data-list.component.ts [hunk 4]
```diff
@@ -53,6 +93,10 @@ export class ReferenceDataListComponent implements OnInit {
     });
   }
 
+  navigateToDeletedItems(): void {
+    this.router.navigate(['deleted-items'], { relativeTo: this.route }).then(() => {});
```
Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-list/reference-data-list.component.ts:97]:
```suggestion
    this.router.navigate(['deleted-items'], { relativeTo: this.route });
```

File: src/app/features/reference-data/reference-data-list/reference-data-list.component.ts [hunk 5]
```diff
@@ -33,6 +40,39 @@ export class ReferenceDataListComponent implements OnInit {
...
+          return of(false);
+        })
+      )
+      .subscribe({
+        next: res => {
+          if (res) {
+            return of(true);
+          }
+          return of(false);
+        }
```
Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-list/reference-data-list.component.ts:74]:
```suggestion
        next: res => of(!!res)
```

File: src/app/features/reference-data/reference-data-list/reference-data-list.component.ts [hunk 6]
```diff
@@ -33,6 +40,39 @@ export class ReferenceDataListComponent implements OnInit {
...
+            map(array =>
+              array.data.map(item => {
+                if (item.reason !== undefined) {
+                  this.disabled = false;
+                }
+              })
+            )
+          )
+        ),
+        take(1),
```
Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-list/reference-data-list.component.ts:59]:
```suggestion
```
shouldn't need two `take(1)` got one on like 46 already

**Review by gjulien-bjss — APPROVED** [review]

**Review by owen-corrigan-bjss — APPROVED** [review]

**Pull request merged**